doc/adrs/adr-render-variables.adoc

```diff
@@ -0,0 +1,35 @@
+= ADR-XXX - Add support for additional variables during representation rendering
+
+== Context
+
+When rendering a representation, Sirius Components will invoke many different user-supplied functions.
+The execution context of these functions is controlled by the corresponding `IComponent` through the `VariableManager` passed to the function.
+The only variables available to the user-supplied functions are the ones defined by Sirius itself.
+
+As a consequence, every invocation of such a function starts "from scratch" and needs to compute everything it needs from only the Sirius-supplied variables each time.
+During a given rendering of a representation, the state of the underlying semantic model, the previous state of the representation (if there is one) and the representation description itself can be assumed to be fixed.
+
+=> opportunity to pre-compute some global information once (per render) to speed up the repeated invocations of the functions.
+
+== Decision
+
+We will add:
+
+```java
+interface IRepresentationDescription
+  default Function<VariableManager, VariableManager> getRenderVariablesProvider() {
+    return (variableManager) -> new VariableManager();
+  }
+}
+```
```

Comment:
Just a couple things to keep in mind when we will move forward with this ADR and its implementation. We don't really need to custom every representation description like that, we can simply add an independent service like that:

```
public interface IRenderingVariableManagerProvider {
  VariableManager get(VariableManager variableManager, ...);
}
```

Which can be injected using Spring at the desired location. Since we try not to add more stuff on the description if it's not required and since it's not strictly necessary in projects like `sirius-components-diagrams` etc.

We will also have to think about the various variables available for the provider but we may have to change the lifecycle of some existing variables like:

- previous representation
- view creation request
- view deletion request
- representation context
- editing context

To start working really on this ADR, we should probably also document the variables which should be available in the rendering VariableManager of the various representations and answer some questions like "should `viewCreationRequests` be a `props` or a variable of `DiagramComponent`" and why :)